### DISCUSSION TITLE
Rerun action but with latest code

### DISCUSSION THREAD

**Discussion by gr2m** [community/community#9025] [Actions]
Creating and testing actions can be a slow process. The usual process is

1. Push code for the workflow
2. Create the event that is triggering the workflow
3. If it fails, fix and push, back to 1

This is how this flow looks like:

<img width="544" alt="image" src="https://user-images.githubusercontent.com/39992/146611571-1a4fdc03-4db9-4d80-be42-51ca015f3a9a.png">

Often times creating the event is not as simple as removing & adding a label. 

What would be great if I could do that instead

1. Push code for the workflow
2. Create the event that is triggering the workflow
3. If it fails, fix and push
4. Run the build again using the latest code. If it fails, back to 3.

It could either be an option to "rerun all jobs using latest code" or add a way for me re-trigger an event, similar to what I can do today with GitHub Apps using the `Redeliver` button

<img width="944" alt="image" src="https://user-images.githubusercontent.com/39992/146611940-51a090a3-eda7-45ca-90f2-be64ae245717.png">

I think adding a `Redeliver the same event` button might be the simplest and most clear.

**Comment by umarcor** [comment]
@gr2m are you aware of [workflow_dispatch](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/)?

![](https://i2.wp.com/user-images.githubusercontent.com/1865328/86147571-2de93700-babf-11ea-8a08-e4beffd3abe9.png?ssl=1)

**Reply by gr2m** [reply]
Yes. It's not the same thing. In my example above, I want the `issues.labeled` event to be retriggered. `workflow_dispatch` does not help me

**Comment by ethomson** [comment]
Hi @gr2m - I strongly agree.  The inner loop for developing workflows and actions is Not Great.  This is something that I'd like to improve in the new year.  In the meantime, sorry that this is painful.

**Reply by gr2m** [reply]
ping me any time if you have more questions, or need more pain points 🤣

**Comment by MiguelRipoll23** [comment]
Love the idea, right now it’s painful to test workflows.

**Comment by pmbanugo** [comment]
Was there any decision on this request? 👀

**Comment by YukiSuter** [comment]
Any update? This can get quite annoying when you have your action set to distinguish between workflow_dispatch and another event!

**Reply by ThosRTanner** [reply]
yes, please. i'm trying to test a repository-dispatch workflow and it's a very non intuitive having to rerun the event in the original repository in order to test the event in the target repository
